Fix a case where  An entry with the same key already exists is throws during installation

yishaigalatzer opened the pull request:
Fix https://github.com/NuGet/Home/issues/1412

A server (or package) might incorrectly return multiple dependencies in the same package.
Although this can be ignored using distinct. It indicates that something is very wrong with the package.

We know of at least one server that combines dependencies, which leads to unrelated dependencies and versions in the same dependency group.

Hence the fix is to throw a better error message.

yishaigalatzer commented:
CC @emgarten @MeniZalzman @deepakaravindr @RanjiniM

deepakaravindr commented:
:shipit:

MeniZalzman commented on src/NuGet.Core/NuGet.Resolver/Strings.resx:
```diff
@@ -123,6 +123,9 @@
     <value>constraint</value>
   </data>
+  <data name="DuplicateDependencyIdsError" xml:space="preserve">
+    <value>The package `{0}` version `{1}` declared a duplicate dependency `{2}`. This might mean the package is corrupt or the server metadata for the package is corrupt.</value>
```
Can we add resources at this time?

emgarten replied:
Other strings use `'` instead of ```

yishaigalatzer replied:
this is for 3.4, will fix the ` -> '

emgarten commented:
:shipit:

Commit:
Fix https://github.com/NuGet/Home/issues/1412

A server (or package) might incorrectly return multiple dependencies in the same package.
Although this can be ignored using distinct. It indicates that something is very wrong with the package.

We know of at least one server that combines dependencies, which leads to unrelated dependencies and versions in the same dependency group.

Hence the fix is to throw a better error message.

yishaigalatzer commented:
Merged through https://github.com/NuGet/NuGet.Client/commit/1bd6cb2b1e0d7f2a28d29d9a9501442ad870d733
